Sources/Lifecycle/Lifecycle.swift

```diff
@@ -357,7 +357,9 @@ public class ComponentLifecycle: LifecycleTask {
     // MARK: - private
 
     private func _start(on queue: DispatchQueue, tasks: [LifecycleTask], callback: @escaping (Error?) -> Void) {
-        precondition(tasks.count > 0, "invalid number of tasks, must be > 0")
+        if tasks.count == 0 {
+            log(level: .notice, "No task provided.")
+        }
```

Comment:
The order of log messages will be a bit meh here with this change:

> No task provided.
> starting

and the message style is slightly off; we start with lowercase letters and no trailing `.` in messages.

Could you move this below the log("starting")?

Message perhaps `no tasks provided`?